tests/components/daikin/test_config_flow.py

```diff
@@ -105,10 +93,31 @@ async def test_discovery(hass, mock_daikin):
 )
 async def test_device_abort(hass, mock_daikin, s_effect, reason):
     """Test device abort."""
-    flow = init_config_flow(hass)
     mock_daikin.factory.side_effect = s_effect
 
-    result = await flow.async_step_user({CONF_HOST: HOST})
+    result = await hass.config_entries.flow.async_init(
+        "daikin", context={"source": SOURCE_USER}, data={CONF_HOST: HOST, KEY_MAC: MAC},
+    )
     assert result["type"] == RESULT_TYPE_FORM
     assert result["errors"] == {"base": reason}
     assert result["step_id"] == "user"
+
+
+@pytest.mark.parametrize(
+    "source, data, unique_id", [(SOURCE_DISCOVERY, {KEY_IP: HOST, KEY_MAC: MAC}, MAC)],
+)
+async def test_discovery(hass, mock_daikin, source, data, unique_id):
+    """Test discovery/zeroconf step."""
+    result = await hass.config_entries.flow.async_init(
+        "daikin", context={"source": source}, data=data,
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    MockConfigEntry(domain="daikin", unique_id=unique_id).add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        "daikin", context={"source": source}, data=data,
+    )
+
+    assert result["type"] == RESULT_TYPE_ABORT
+    assert result["reason"] == "already_in_progress"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)